File: node.py
class Node:
  def __init__(self, letter=None):
    self.children = []
    self.endWord = False
    self.letter = letter

  def __str__(self):
    return str(self.letter)

  def addChild(self, child):
    self.children.append(child)

  def getChildAt(self, index):
    return self.children[index]

  def getChildWithValue(self, letter):
    for child in self.children:
      if child.letter == letter:
        return child
    return None

  def hasChild(self, letter):
    for child in self.children:
      if child.letter == letter:
        return True
    return False

File: js/node.js
var Node = function(letter) {
  this.children = [];
  this.endWord = false;
  this.letter = letter;
}

Node.prototype.addChild = function(child) {
  this.children.push(child);
}

Node.prototype.getChildWithValue = function(letter) {
  for (i in this.children) {
    if (this.children[i].letter == letter) {
      return this.children[i];
    }
  }
  return null;
}